source/gui/settingsDialogs.py

```diff
@@ -2530,21 +2530,21 @@ def __init__(self, parent):
 		# Translators: This is the label for a combo box for selecting a
 		# method of detecting changed content in terminals in the advanced
 		# settings panel.
-		# Choices are automatic, character, and line.
-		diffAlgoComboText = _("C&hange detection preference:")
+		# Choices are automatic, prefer Diff Match Patch, and prefer Difflib.
+		diffAlgoComboText = _("D&iff algorithm:")
 		diffAlgoChoices = [
 			# Translators: A choice in a combo box in the advanced settings
 			# panel to have NVDA determine the method of detecting changed
 			# content in terminals automatically
-			pgettext("change detection preference", "automatic (line)"),
+			_("Automatic (Difflib)"),
 			# Translators: A choice in a combo box in the advanced settings
 			# panel to have NVDA detect changes in terminals
-			# by character when supported.
-			pgettext("change detection preference", "character"),
+			# by character when supported, using the diff match patch algorithm.
+			_("Prefer Diff Match Patch"),

```

Comment:
I think this is a little bit misleading right? Really enabling this means use where available / possible? DiffLib is always the fallback option. Where as DMP is never the fallback option.

Reply:
Nope, this code would *always* use DMP regardless of user preference:

```python
class Subclass(LiveText):
    def _get_diffAlgo(self):
        from diffHandler import dmp
        return dmp
```

Changing `dmp` to `difflib` in the above would forcibly use difflib regardless of the user setting.